harness/harness-core-ui
# fix: [CDS-84514]: fixed create step  breaking page issue after reopening saved step which has multiselect type fields

### Summary
Issue - Create step page is breaking after reopening saved step which has multiselect type fields.

Solution - Earlier we were using 2 different components for rendering multiType fields in create form and deployment form and we were also manipulating initialValues which manipulates the formik values. For create form case, string value was getting passed to the multiselect component, now we reused the usage of 2 different component and used only one. Passed custom value to it which basically manipulates comma-separated string to array of objects.

Manual testing - 

Create Form
- New fields are getting added.
- Values added to fields are retained upon reopening the form.
- Runtime inputs are working fine for all type of fields.
- Expression inputs are working fine for all type of fields.
- Making changes to field are getting reflected to yaml.

Run form

- Field changes are getting reflected in yaml.
- Upon re-running the pipeline, fields are getting proper values in proper format.

Templates

- Field changes are getting reflected in yaml.
- After savings we dont see 'unsaved changes'.




#### Screenshots

Before - 

https://github.com/harness/harness-core-ui/assets/96036463/cb9ce06b-8e10-4b41-bc91-ffffcff792ff


After - 

https://github.com/harness/harness-core-ui/assets/96036463/12401f21-5dd6-4f25-8cd5-adffcbecc34b





#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)
